airflow/providers/apache/hive/hooks/hive.py

```diff
@@ -709,25 +707,20 @@ def _get_max_partition_from_part_specs(
         # Assuming all specs have the same keys.
         if partition_key not in part_specs[0].keys():
             raise AirflowException(f"Provided partition_key {partition_key} is not in part_specs.")
-        is_subset = None
-        if filter_map:
-            is_subset = set(filter_map.keys()).issubset(set(part_specs[0].keys()))
-        if filter_map and not is_subset:
+        if filter_map and not set(filter_map).issubset(set(part_specs[0])):
```

Comment:
```suggestion
        if filter_map and not set(filter_map).issubset(part_specs[0]):
```

The set cast is done automatically.